# 🛒 ¿Qué hay en las estanterías de La Habana?

## Análisis económico del origen de productos en establecimientos comerciales

Este trabajo presenta un análisis económico descriptivo de la economía cubana a partir del origen de los productos comercializados en La Habana. El estudio utiliza la presencia de productos nacionales e importados como un indicador para analizar la estructura económica actual, el comercio y la dependencia externa.

📊 Proyecto de Ciencia de Datos | Enero 2026

## 📍 Contexto económico del comercio en Cuba

El análisis del comercio minorista permite observar de manera concreta el funcionamiento de la economía. En Cuba, la circulación de bienes se realiza a través de distintos tipos de establecimientos, cada uno con un papel específico dentro del sistema económico.

### 🏪 MiPyMEs (Micro, Pequeñas y Medianas Empresas)
Establecimientos privados autorizados desde 2021. Venden productos alimenticios, de aseo, ropa, etc. Ejemplos: bodegas privadas, tiendas, minimercados.

### 🏢 Tiendas Estatales
Administradas por el gobierno. Incluyen:
- Tiendas en MLC (moneda libremente convertible)
- TRD (Tiendas de Recaudación de Divisas)
- Tiendas Caribe

### 🛍️ Plataformas Online
Como Revolico, donde particulares ofrecen productos.

---

## 🛒 Productos de Primera Necesidad

Son aquellos productos **básicos e indispensables** para la vida diaria:

**Alimentos básicos:**
- Arroz
- Aceite comestible
- Azúcar
- Frijoles
- Pollo
- Huevos

Estos productos forman parte de la **canasta básica familiar** y su disponibilidad y origen afectan directamente a toda la población.

## 🎯 Enfoque del análisis

Desde mediados del siglo XX, la economía cubana ha transitado por distintos modelos económicos. Productos como el azúcar, el tabaco, el ron y el café desempeñaron históricamente un papel central en la producción y el comercio.

En las últimas décadas, este panorama ha cambiado de forma significativa. El aumento de las importaciones y la disminución de la producción nacional en diversas categorías reflejan transformaciones estructurales en la economía.

Este análisis se enfoca en examinar el grado de dependencia de productos importados en los establecimientos comerciales de La Habana, tanto en bienes básicos como en productos históricamente relevantes para la economía nacional.

In [1]:
import sys
sys.path.append('../src')

from analisis import *
from graficas import *
import warnings
warnings.filterwarnings('ignore')

print("✅ Librerías cargadas")

✅ Librerías cargadas


## 🤔 ¿Qué Crees Tú?

Antes de ver los datos, vamos a hacer una predicción basada en tu experiencia.

Cuando entras a una tienda a en La Habana, **¿qué porcentaje de los productos crees que son de origen cubano?**

In [2]:
print("="*60)
print("  ENCUESTA: ¿Cuál es tu predicción?")
print("="*60)
print()
print("¿Qué porcentaje de productos en tiendas de La Habana")
print("crees que son de ORIGEN NACIONAL (cubano)?")
print()
print("A) Mayoría nacional (más del 60%)")
print("B) Equilibrado (40-60%)")
print("C) Mayoría importado (20-40%)")
print("D) Alta dependencia (menos del 20%)")
print()

respuesta = input("Tu respuesta (A, B, C o D): ").upper()

respuestas = {
    'A': 'Mayoría nacional (más del 60%)',
    'B': 'Equilibrado (40-60%)',
    'C': 'Mayoría importado (20-40%)',
    'D': 'Alta dependencia (menos del 20%)'
}

if respuesta in respuestas:
    print()
    print(f"✓ Has seleccionado: {respuestas[respuesta]}")
    print()
    print("Guardemos esta predicción en mente...")
    print("Ahora vamos a ver qué dicen los datos reales.")
    print("="*60)
else:
    print()
    print("⚠ Respuesta no válida. Por favor selecciona A, B, C o D")

  ENCUESTA: ¿Cuál es tu predicción?

¿Qué porcentaje de productos en tiendas de La Habana
crees que son de ORIGEN NACIONAL (cubano)?

A) Mayoría nacional (más del 60%)
B) Equilibrado (40-60%)
C) Mayoría importado (20-40%)
D) Alta dependencia (menos del 20%)


⚠ Respuesta no válida. Por favor selecciona A, B, C o D


In [3]:
est_mipymes = cargar_establecimientos('../data/mipymes.json')
est_revolico = cargar_establecimientos('../data/revolico.json')
tasas = cargar_tasas_cambio('../data/toque.json')

prod_mipymes = extraer_productos(est_mipymes)
prod_revolico = extraer_productos(est_revolico)

productos = combinar_productos([prod_mipymes, prod_revolico])

total_establecimientos = len(est_mipymes) + len(est_revolico)

print(f"📍 Establecimientos Mipymes: {len(est_mipymes)}")
print(f"📍 Anuncios Revolico: {len(est_revolico)}")
print(f"📍 Total establecimientos: {total_establecimientos}")
print(f"📦 Productos registrados: {len(productos)}")

📍 Establecimientos Mipymes: 37
📍 Anuncios Revolico: 45
📍 Total establecimientos: 82
📦 Productos registrados: 353


## 🔎 Veamos Qué Tan Acertada Fue Tu Predicción

Ahora es momento de ver los resultados reales. ¿Qué porcentaje de productos en las estanterías son realmente de origen cubano?

Los datos están a punto de revelar la respuesta...

In [4]:
origen = contar_por_origen(productos)

print("="*60)
print("RESULTADOS: ORIGEN DE LOS PRODUCTOS")
print("="*60)
print()
print(f"Total de productos: {origen['total']}")
print()
print(f"Nacionales (CU): {origen['nacional']} productos ({origen['porcentaje_nacional']}%)")
print(f"Importados: {origen['importado']} productos ({origen['porcentaje_importado']}%)")
print()
print("="*60)
print()

porcentaje = origen['porcentaje_nacional']

if porcentaje > 60:
    print("Resultado: A - Mayoría nacional")
elif porcentaje >= 40:
    print("Resultado: B - Equilibrado")
elif porcentaje >= 20:
    print("Resultado: C - Mayoría importado")
else:
    print("Resultado: D - Alta dependencia")

RESULTADOS: ORIGEN DE LOS PRODUCTOS

Total de productos: 353

Nacionales (CU): 36 productos (10.2%)
Importados: 317 productos (89.8%)


Resultado: D - Alta dependencia


In [5]:
fig = grafica_pastel_origen(productos)
mostrar_grafica(fig)


### 📊 ¿Qué Significa Este Resultado?

Estos números nos muestran la realidad del comercio en La Habana hoy. La proporción entre productos nacionales e importados refleja cómo ha cambiado la economía cubana en las últimas décadas.

### 🍩 Distribución por Categoría

Esta gráfica de dona muestra qué categorías de productos dominan el mercado. Cada color representa una categoría diferente, y el tamaño del segmento indica cuántos productos de esa categoría encontramos.


In [ ]:
fig = grafica_dona_categorias(productos)
mostrar_grafica(fig)

## 🏭 Productos Que Hicieron Historia

Cuba tiene cuatro productos que marcaron su economía durante generaciones:

**Azúcar:** Durante más de 150 años, Cuba fue una potencia azucarera mundial. Centrales azucareros en toda la isla, miles de empleos, pueblos enteros dedicados a la caña.

**Café:** Las montañas de Oriente producían café reconocido internacionalmente. El café cubano era sinónimo de calidad.

**Ron:** Havana Club, Santiago de Cuba, Legendario... marcas que llevan el nombre de Cuba al mundo.

**Tabaco:** Los puros cubanos siguen siendo considerados los mejores del mundo.

¿Qué queda de esa presencia en las tiendas de hoy?

In [7]:
estrategicos = analizar_productos_estrategicos(productos)

print("="*60)
print("PRODUCTOS ESTRATÉGICOS EN LAS TIENDAS")
print("="*60)

if 'azucar' in estrategicos:
    azucar = estrategicos['azucar']
    print()
    print("AZÚCAR:")
    print(f"  Total: {azucar['total']}")
    print(f"  Nacional: {azucar['nacional']} ({azucar['porcentaje_nacional']}%)")
    print(f"  Importado: {azucar['importado']}")

if 'cafe' in estrategicos:
    cafe = estrategicos['cafe']
    print()
    print("CAFÉ:")
    print(f"  Total: {cafe['total']}")
    print(f"  Nacional: {cafe['nacional']} ({cafe['porcentaje_nacional']}%)")
    print(f"  Importado: {cafe['importado']}")

if 'ron' in estrategicos:
    ron = estrategicos['ron']
    print()
    print("RON:")
    print(f"  Total: {ron['total']}")
    print(f"  Nacional: {ron['nacional']} ({ron['porcentaje_nacional']}%)")
    print(f"  Importado: {ron['importado']}")

if 'tabaco' in estrategicos:
    tabaco = estrategicos['tabaco']
    print()
    print("TABACO:")
    print(f"  Total: {tabaco['total']}")
    print(f"  Nacional: {tabaco['nacional']} ({tabaco['porcentaje_nacional']}%)")
    print(f"  Importado: {tabaco['importado']}")

print()
print("="*60)

PRODUCTOS ESTRATÉGICOS EN LAS TIENDAS

AZÚCAR:
  Total: 22
  Nacional: 0 (0.0%)
  Importado: 22

CAFÉ:
  Total: 51
  Nacional: 9 (17.65%)
  Importado: 42

RON:
  Total: 10
  Nacional: 10 (100.0%)
  Importado: 0

TABACO:
  Total: 12
  Nacional: 12 (100.0%)
  Importado: 0



### 😢 Una Historia de Pérdida

Los números no mienten:

- **Azúcar:** El producto que definió a Cuba por siglos ahora es mayormente importado
- **Café:** Presión creciente de marcas extranjeras
- **Ron:** Mantiene presencia 100% nacional, producto de exportación reconocido mundialmente
- **Tabaco:** Bastión de la producción nacional, los puros cubanos siguen siendo únicos en el mercado

¿Cómo pasamos de productores en productos como la **Azúcar** y el **Café** a consumidores?

### 🔎 Análisis de los Resultados

Cada uno de estos productos cuenta una historia diferente:

- Algunos mantienen fuerte presencia nacional
- Otros han sido desplazados por importaciones
- Algunos prácticamente desaparecieron del mercado local

Estos números reflejan décadas de cambios en la economía cubana: desde el Período Especial de los 90s hasta las reformas más recientes.

## 🍚 La Canasta Básica

Más allá de los productos históricos, ¿qué pasa con lo básico? 

Arroz, aceite, frijoles... ¿De dónde vienen estos productos básicos?


In [8]:
necesidad = analizar_primera_necesidad(productos)

print("="*60)
print("PRODUCTOS DE PRIMERA NECESIDAD")
print("="*60)

for producto in necesidad:
    datos = necesidad[producto]
    dependencia = 100 - datos['porcentaje_nacional']
    print()
    print(f"{producto.upper()}:")
    print(f"  Total encontrado: {datos['total']}")
    print(f"  Nacional: {datos['nacional']} ({datos['porcentaje_nacional']}%)")
    print(f"  Importado: {datos['importado']}")
    print(f"  Dependencia: {dependencia:.1f}%")

print()
print("="*60)



PRODUCTOS DE PRIMERA NECESIDAD

ARROZ:
  Total encontrado: 34
  Nacional: 1 (2.94%)
  Importado: 33
  Dependencia: 97.1%

POLLO:
  Total encontrado: 19
  Nacional: 0 (0.0%)
  Importado: 19
  Dependencia: 100.0%

ACEITE:
  Total encontrado: 22
  Nacional: 0 (0.0%)
  Importado: 22
  Dependencia: 100.0%

FRIJOL:
  Total encontrado: 20
  Nacional: 0 (0.0%)
  Importado: 20
  Dependencia: 100.0%



### ⚠️ Dependencia en lo Básico

Los productos más esenciales para la alimentación diaria muestran alta dependencia de importaciones.

Esto significa que el abastecimiento de productos básicos depende del comercio internacional. Cualquier interrupción afecta directamente a las familias.


## 🌍 ¿De Dónde Vienen los Productos Importados?

Ya sabemos que la mayoría de productos son importados. Pero, ¿de qué países vienen?

In [9]:
paises = contar_productos_importados(productos)

print("="*60)
print("PRINCIPALES PAÍSES DE ORIGEN")
print("="*60)

for i in range(min(10, len(paises))):
    pais = paises[i]
    numero = i + 1
    nombre_pais = pais['pais']
    cantidad = pais['cantidad']
    print(str(numero) + ". " + nombre_pais + " - " + str(cantidad) + " productos")

print("="*60)

PRINCIPALES PAÍSES DE ORIGEN
1. No especificado - 165 productos
2. BR - 36 productos
3. ES - 36 productos
4. US - 22 productos
5. MX - 11 productos
6. ITA - 11 productos
7. MA - 9 productos
8. COL - 6 productos
9. TR - 6 productos
10. DO - 3 productos


In [10]:
fig = grafica_paises(productos, top=10)
mostrar_grafica(fig)

### 🗺️ Observación sobre los Orígenes

Los productos llegan de múltiples países: Brasil, España, México, y sorprendentemente, también de Estados Unidos.

A pesar del embargo comercial vigente desde 1962, algunos productos estadounidenses sí llegan a Cuba, principalmente alimentos y medicinas (categorías con excepciones específicas en el embargo).

La presencia de productos de países lejanos implica mayores costos de transporte y tiempos de envío más largos.

## 💰 ¿Qué Cuesta Más: Lo Nacional o Lo Importado?

Una pregunta importante: ¿los productos cubanos son más baratos o más caros que los importados?


In [11]:
precios = comparar_precios(productos)

print("="*60)
print("COMPARACIÓN DE PRECIOS (TODOS LOS PRODUCTOS)")
print("="*60)
print()
print(f"Precio promedio NACIONAL:  {precios['promedio_nacional']:.2f} CUP")
print(f"Precio promedio IMPORTADO: {precios['promedio_importado']:.2f} CUP")
print()
print(f"Diferencia: {precios['diferencia']:.2f} CUP")
print(f"Porcentaje: {precios['porcentaje_diferencia']:.1f}%")
print()
print("="*60)

if precios['diferencia'] > 0:
    print()
    print("→ Los productos importados son más caros")
else:
    print()
    print("→ Los productos nacionales son más caros")

COMPARACIÓN DE PRECIOS (TODOS LOS PRODUCTOS)

Precio promedio NACIONAL:  16740.28 CUP
Precio promedio IMPORTADO: 1445.11 CUP

Diferencia: -15295.17 CUP
Porcentaje: -91.4%


→ Los productos nacionales son más caros


In [12]:
fig = grafica_precios_categoria(productos)
mostrar_grafica(fig)

### 🤔 Un Dato Engañoso

Los datos anteriores muestran que los productos nacionales parecen mucho más caros. Pero hay un problema: **el alcohol y el tabaco** cubanos son productos premium de exportación con precios muy elevados que distorsionan el promedio.

Para obtener una imagen más realista del mercado cotidiano, necesitamos analizar los precios **sin** estos productos de lujo.


In [13]:
print("="*60)
print("COMPARACIÓN DE PRECIOS (SIN ALCOHOL NI TABACO)")
print("="*60)
print()

precios_basicos = comparar_precios_sin_lujos(productos)

print(f"Productos analizados:")
print(f"  Nacional: {precios_basicos['cantidad_nacional']}")
print(f"  Importado: {precios_basicos['cantidad_importado']}")
print()
print(f"Precio promedio NACIONAL:  {precios_basicos['promedio_nacional']:.2f} CUP")
print(f"Precio promedio IMPORTADO: {precios_basicos['promedio_importado']:.2f} CUP")
print()
print(f"Diferencia: {precios_basicos['diferencia']:.2f} CUP")
print(f"Porcentaje: {precios_basicos['porcentaje_diferencia']:.1f}%")
print()
print("="*60)

if precios_basicos['diferencia'] > 0:
    print()
    print("→ Los productos importados son más caros")
else:
    print()
    print("→ Los productos nacionales son más caros")

COMPARACIÓN DE PRECIOS (SIN ALCOHOL NI TABACO)

Productos analizados:
  Nacional: 11
  Importado: 317

Precio promedio NACIONAL:  1704.55 CUP
Precio promedio IMPORTADO: 1445.11 CUP

Diferencia: -259.44 CUP
Porcentaje: -15.2%


→ Los productos nacionales son más caros


### 📊 Una Mejor Perspectiva

El **promedio** puede ser engañoso cuando hay valores extremos. La **mediana** (el valor del medio) nos da una imagen más real del precio típico.

In [14]:
estadisticas = analizar_estadisticas_precios(productos)

print()
print("="*60)
print("ESTADÍSTICAS DE PRECIOS COMPLETAS")
print("="*60)
print()
print("PRODUCTOS NACIONALES:")
print(f"  Media (Promedio): {estadisticas['nacional']['media']:.2f} CUP")
print(f"  Mediana:          {estadisticas['nacional']['mediana']:.2f} CUP")
print(f"  Moda:             {estadisticas['nacional']['moda']:.2f} CUP")
print(f"  Mínimo:           {estadisticas['nacional']['minimo']:.2f} CUP")
print(f"  Máximo:           {estadisticas['nacional']['maximo']:.2f} CUP")
print()
print("PRODUCTOS IMPORTADOS:")
print(f"  Media (Promedio): {estadisticas['importado']['media']:.2f} CUP")
print(f"  Mediana:          {estadisticas['importado']['mediana']:.2f} CUP")
print(f"  Moda:             {estadisticas['importado']['moda']:.2f} CUP")
print(f"  Mínimo:           {estadisticas['importado']['minimo']:.2f} CUP")
print(f"  Máximo:           {estadisticas['importado']['maximo']:.2f} CUP")



ESTADÍSTICAS DE PRECIOS COMPLETAS

PRODUCTOS NACIONALES:
  Media (Promedio): 16740.28 CUP
  Mediana:          1750.00 CUP
  Moda:             2000.00 CUP
  Mínimo:           250.00 CUP
  Máximo:           365500.00 CUP

PRODUCTOS IMPORTADOS:
  Media (Promedio): 1445.11 CUP
  Mediana:          750.00 CUP
  Moda:             750.00 CUP
  Mínimo:           36.00 CUP
  Máximo:           24600.00 CUP




## Por Qué Cinco Medidas en Lugar de Una

Calculé **media, mediana, moda, mínimo y máximo** porque una sola medida habría generado conclusiones incorrectas sobre este mercado.

**El problema:** Nuestros datos tienen productos desde 425 CUP (pastas) hasta 365,500 CUP (rones premium). La media de productos nacionales es 16,740 CUP, pero esto **no representa** el precio típico que encontrará un consumidor.

### Decisión Metodológica

Cada medida cumple un propósito específico en este análisis:

- **Media:** Cuantifica el impacto total del mercado, incluyendo segmentos premium
- **Mediana:** Identifica el precio real del producto típico, sin distorsión por extremos
- **Moda:** Revela el punto de precio más común, indicando concentración de oferta
- **Mínimo/Máximo:** Delimitan el rango de accesibilidad del mercado

### Hallazgos que Justifican el Enfoque

La diferencia entre media (16,740 CUP) y mediana en productos nacionales evidenció que **el promedio estaba distorsionado** por rones y tabacos de exportación. Sin la mediana, habría concluido erróneamente que los productos cubanos son sistemáticamente más caros.

Esta distorsión motivó el análisis adicional **"sin productos de lujo"**, revelando que en categorías básicas la diferencia de precios es mucho menor.

En mercados segmentados como este, donde coexisten productos de primera necesidad con artículos de exportación premium, **múltiples medidas son necesarias** para capturar la realidad completa del comportamiento de precios.


### 📈 Visualización en Áreas

Esta gráfica de áreas apiladas permite ver mejor la diferencia acumulada de precios entre productos nacionales e importados a través de las categorías.


In [15]:
fig = grafica_area_precios_comparacion(productos)
mostrar_grafica(fig)

### 💸 El Factor Precio

Excluyendo alcohol y tabaco, vemos una realidad diferente. Los precios tienen explicaciones complejas:

- Economías de escala menores en producción local
- Costos de insumos (muchos también importados)

El precio es solo uno de varios factores. La **disponibilidad** y la **calidad** también influyen en las decisiones de compra.



## 📈 Tendencias por Categoría
¿Hay categorías donde los productos nacionales dominan? ¿O todas muestran alta dependencia de importaciones?

In [16]:
fig = grafica_linea_tendencia(productos)
mostrar_grafica(fig)

## 🌎 Accesibilidad: Cuba vs El Mundo

Los precios en CUP no cuentan toda la historia. Para entender el verdadero impacto económico, debemos comparar con el **poder adquisitivo**.

Usando las tasas de cambio de **El Toque**, podemos ver qué  representa cada producto tanto para un cubano como para alguien en otro país.


In [17]:
print("="*60)
print("TASAS DE CAMBIO - EL TOQUE")
print("="*60)
print()
print(f"Fuente: {tasas['source']}")
print(f"Actualización: {tasas['last_update']}")
print()

for tasa in tasas['purchase_rates']:
    moneda = tasa['currency']
    valor = tasa['purchase']
    print(f"  {moneda}: {valor} CUP")

print()
print("="*60)

TASAS DE CAMBIO - EL TOQUE

Fuente: El Toque
Actualización: 2025-11-15T23:47:42.231838

  USD: 460 CUP
  EUR: 500 CUP
  MLC: 220 CUP
  CUP: 100 CUP



In [18]:
mas_caros = productos_mas_caros(productos, 5)

print("="*60)
print("CONVERSIÓN DE PRECIOS (TOP 5 MÁS CAROS)")
print("="*60)
print()

for prod in mas_caros:
    precio_cup = prod['precio']
    precio_usd = convertir_precio(precio_cup, tasas, 'USD')
    precio_eur = convertir_precio(precio_cup, tasas, 'EUR')
    precio_mlc = convertir_precio(precio_cup, tasas, 'MLC')
    
    print(f"{prod['nombre']}")
    print(f"  CUP: {precio_cup:.2f}")
    print(f"  USD: {precio_usd:.2f}")
    print(f"  EUR: {precio_eur:.2f}")
    print(f"  MLC: {precio_mlc:.2f}")
    print()

print("="*60)

CONVERSIÓN DE PRECIOS (TOP 5 MÁS CAROS)

Ron cubano
  CUP: 365500.00
  USD: 794.57
  EUR: 731.00
  MLC: 1661.36

Ron añejo 15 años sellado en estuche original
  CUP: 51600.00
  USD: 112.17
  EUR: 103.20
  MLC: 234.55

Ron Habana Club 15 años en cilindro
  CUP: 43000.00
  USD: 93.48
  EUR: 86.00
  MLC: 195.45

Mareva Tabacos habanos de todos los modelos
  CUP: 34400.00
  USD: 74.78
  EUR: 68.80
  MLC: 156.36

Ron Santiago 65 aniversario Edición Limitada
  CUP: 30100.00
  USD: 65.43
  EUR: 60.20
  MLC: 136.82



## 🏭 Marcas y Fabricantes

¿Qué fabricantes cubanos mantienen presencia en el mercado? ¿Qué marcas extranjeras encontramos?

In [19]:
fab_nacionales = top_fabricantes_nacionales(productos, 10)

print("="*60)
print("TOP 10 FABRICANTES NACIONALES")
print("="*60)

for i, fab in enumerate(fab_nacionales, 1):
    print(f"{i:>2}. {fab['fabricante']:<40} {fab['cantidad']:>3} productos")

print("="*60)


TOP 10 FABRICANTES NACIONALES
 1. Varios                                    10 productos
 2. Habana Club                                7 productos
 3. Isla Grande                                5 productos
 4. Cohiba                                     2 productos
 5. Mareva                                     2 productos
 6. Santiago                                   2 productos
 7. TropiGusto                                 1 productos
 8. Delicias Estrella                          1 productos
 9. Minino                                     1 productos
10. Paraná                                     1 productos


In [20]:
fig = grafica_fabricantes(productos, top=10)
mostrar_grafica(fig)

### 🏷️ Marcas que Dominan el Mercado

Entre los fabricantes nacionales, vemos nombres históricos como Habana Club y Cohiba, principalmente en categorías de exportación (ron y tabaco).

Los fabricantes extranjeros tienen presencia diversificada en múltiples categorías, desde alimentos básicos hasta productos procesados.

In [21]:
fab_importados = top_fabricantes_importados(productos, 10)

print("="*60)
print("TOP 10 FABRICANTES EXTRANJEROS")
print("="*60)

for i, fab in enumerate(fab_importados, 1):
    print(f"{i:>2}. {fab['fabricante']:<40} {fab['cantidad']:>3} productos")

print("="*60)


TOP 10 FABRICANTES EXTRANJEROS
 1.                                          146 productos
 2. Vima                                      16 productos
 3. Del Campo                                 11 productos
 4. San Diego                                  5 productos
 5. Energy                                     4 productos
 6. Sublime                                    4 productos
 7. Doga                                       4 productos
 8. La LLave                                   4 productos
 9. TWINNIES                                   4 productos
10. La Llave                                   4 productos


## 📊 Síntesis de Resultados

Después de analizar los datos de establecimientos comerciales en La Habana, estos son los hallazgos principales:

### 1. Dependencia de Importaciones
La mayoría de productos en las tiendas son importados, lo que evidencia un cambio significativo en el modelo económico cubano.

### 2. Productos Estratégicos
Los productos que históricamente definieron la economía (azúcar, café, ron, tabaco) muestran diferentes niveles de presencia nacional. Algunos mantienen su posición, otros no.

### 3. Productos Básicos
Los alimentos de primera necesidad (arroz, aceite, frijoles) dependen considerablemente de importaciones.

### 4. Origen Diverso
Los productos vienen de múltiples países, incluyendo algunos de Estados Unidos a pesar del embargo.

### 5. Fabricantes
Existen fabricantes nacionales activos, pero las marcas extranjeras tienen fuerte presencia en múltiples categorías.


## 🔮 Reflexión final

El análisis del origen de los productos comercializados en La Habana permite observar de forma concreta el estado actual de la economía cubana. Los datos reflejan una transformación significativa respecto a modelos económicos anteriores, con una fuerte dependencia de las importaciones.

Esta situación plantea interrogantes económicas relevantes:
- Sostenibilidad del modelo actual
- Vulnerabilidad ante interrupciones del comercio
- Posibilidades de fortalecimiento de la producción nacional

Comprender la estructura económica actual es un paso esencial para cualquier análisis futuro. Los datos están disponibles; el análisis continúa.

## 📚 Referencias y Metodología
**Autora:** Mariam Parra Cárdenas  
**Grupo:** D-111  
**Facultad:** Matemática y Computación  
**Universidad de La Habana**

**Fuentes de datos:**
- 30+ MiPyMEs visitadas presencialmente
- Anuncios de Revolico.com
- El toque

**Período:** Noviembre-Diciembre 2025

**Ubicación:** La Habana, Cuba

**Metodología:** Análisis descriptivo de datos estructurados en JSON, sin uso de librerías externas como pandas.

**Código:** Todo el código de análisis y visualización está disponible en el repositorio del proyecto.

---

**Fin del análisis** 📊